src/utils/daos.c

```diff
@@ -1082,152 +1131,164 @@ help_hdlr(struct cmd_args_s *ap)
-"resources:\n"
-"	  pool             pool\n"
-"	  container (cont) container\n"
-"	  version          print command version\n"
-"	  help             print this message and exit\n");
-
-	fprintf(stream, "\n"
-"pool commands:\n"
-"	  list-containers  list all containers in pool\n"
-"	  list-cont\n"
-"	  query            query a pool\n"
-"	  stat             get pool statistics\n"
-"	  list-attrs       list pool user-defined attributes\n"
-"	  get-attr         get pool user-defined attribute\n");
-
-	fprintf(stream,
-"pool options:\n"
-"	--pool=UUID        pool UUID\n"
-"	--sys-name=STR     DAOS system name context for servers (\"%s\")\n"
-"	--sys=STR\n"
-"	--svc=RANKS        pool service replicas like 1,2,3\n"
-"	--attr=NAME        pool attribute name to get\n",
-	default_sysname);
-
-	fprintf(stream, "\n"
-"container (cont) commands:\n"
-"	  create           create a container\n"
-"	  destroy          destroy a container\n"
-"	  list-objects     list all objects in container\n"
-"	  list-obj\n"
-"	  query            query a container\n"
-"	  get-prop         get a container's properties\n"
-"	  set-prop         set a container's properties\n"
-"	  get-acl          get a container's ACL\n"
-"	  overwrite-acl    replace a container's ACL\n"
-"	  update-acl       add/modify entries in a container's ACL\n"
-"	  delete-acl       delete an entry from a container's ACL\n"
-"	  set-owner        change the user and/or group that own a container\n"
-"	  stat             get container statistics\n"
-"	  list-attrs       list container user-defined attributes\n"
-"	  del-attr         delete container user-defined attribute\n"
-"	  get-attr         get container user-defined attribute\n"
-"	  set-attr         set container user-defined attribute\n"
-"	  create-snap      create container snapshot (optional name)\n"
-"			   at most recent committed epoch\n"
-"	  list-snaps       list container snapshots taken\n"
-"	  destroy-snap     destroy container snapshots\n"
-"			   by name, epoch or range\n"
-"	  rollback         roll back container to specified snapshot\n");
-
+	if (argc == 2) {
+		FIRST_LEVEL_HELP
+	} else if (strcmp(argv[2], "pool") == 0) {
+		fprintf(stream, "\n"
+		"pool commands:\n"
+		"	  list-containers  list all containers in pool\n"
+		"	  list-cont\n"
+		"	  query            query a pool\n"
+		"	  stat             get pool statistics\n"
+		"	  list-attrs       list pool user-defined attributes\n"
+		"	  get-attr         get pool user-defined attribute\n");
+		
+		fprintf(stream,
+		"pool options:\n"
+		"	--pool=UUID        pool UUID\n"
+		"	--sys-name=STR     DAOS system name context for servers (\"%s\")\n"
+		"	--sys=STR\n"
+		"	--svc=RANKS        pool service replicas like 1,2,3\n"
+		"	--attr=NAME        pool attribute name to get\n",
+			default_sysname);
+
+	} else if (strcmp(argv[2], "container") == 0 ||
+		   strcmp(argv[2], "cont") == 0) {
+		if (argc == 3) {
+			ALL_CONT_CMDS_HELP
+		} else if (strcmp(argv[3], "create") == 0) {
 #if 0
```

Comment:
Might be a good time to remove this stuff that's #if'd out. Is there a reason this has been left in?

Reply:
Well, I left this in place because it was here and I don't know the original reason for this lines to be kept. But I am sure @kccain will tell us why and what to do, as this comes from one of his previous commit (e54c143c84) ;-)

Reply:
No idea why I inserted the #if 0 in that commit, so if it's alright while we're making this change, I would recommend removing it.